error.py

```diff
@@ -42,10 +42,10 @@ def __init__(self, url):
 class BadRequestError(Exception):
     '''Exception raised when the API call was not completed successfully'''
 
-    def __init__(self, status_code):
+    def __init__(self, response):
         super(BadRequestError, self).__init__()
-        self.message = "The API call was unsuccessful with status code: " + \
-            str(status_code)
+        self.message = "The API call was unsuccessful with response:\n" + \
+            str(response)
```

Comment:
What type of object is `response`? Will it be easily readable in the error message? Is not, lets revert back to showing the error code only. Or in cases where there is an error message, we could print that.

Reply:
Here is what the error object would look like 
```
{
  "responses": [
    {
      "error": {
        "code": 3,
        "message": " unsupported URL specified etc"  
      }
    }
  ]
}
```

Reply:
in the case of no error it would be a bit more complex but since we only worry about the scenario where we have an error we use this !

Reply:
Showing an entire object in the message might not be good. Let's either revert to error status code (as was earlier) or use `responses.error.message`

Reply:
sounds good. will update accordingly!